src/Compiler/Service/FSharpCheckerResults.fs

```diff
@@ -1079,14 +1079,16 @@ type internal TypeCheckInfo
 
     /// Gets all field identifiers of a union case that can be referred to in a pattern.
     let GetUnionCaseFields caseIdRange alreadyReferencedFields =
+        let alreadyReferencesSet = alreadyReferencedFields |> Set.ofList
```

Comment:
We will be allocating an additional set here for every single union case here, what are allocation differences for typechecker here?

Reply:
But the lambda does efficiently for over CapturedNameResolutions O(n) times List.contains alreadyReferenced O(n) (each item through) so O(n * n) vs with a Set it's O(n*log n)

Reply:
Sure, but memory and gc oftentimes has much more significant impact, especially in long-running tooling scenarios. Especially given that sizes of these lists are not that huge.

So, I'm more worried that we'll stop-the-world GC more often than we're searching slower.

Reply:
Is there a way to test your worry?